src/Microsoft.VisualStudio.Threading/ThreadingTools.cs

```diff
@@ -110,6 +110,30 @@ public static SpecializedSyncContext Apply(this SynchronizationContext syncConte
             return SpecializedSyncContext.Apply(syncContext, checkForChangesOnRevert);
         }
 
+        /// <summary>
+        /// Wraps a task with one that will complete as cancelled based on a cancellation token,
+        /// allowing someone to await a task but be able to break out early by cancelling the token.
+        /// </summary>
+        /// <param name="task">The task to wrap.</param>
+        /// <param name="cancellationToken">The token that can be canceled to break out of the await.</param>
+        /// <returns>The wrapping task.</returns>
+        internal static Task WithCancellationJtfAware(this Task task, CancellationToken cancellationToken)
```

Comment:
I like your idea of the renamed overload. I've applied that change.
I don't see how using `ContinueWith` instead of `await` would avoid needing to special case. `ContinueWith` needs a `TaskScheduler` to use, and I'd have to choose whether to use `TaskScheduler.Default` or `TaskScheduler.FromCurrentSynchronizationContext`.